### ISSUE TITLE
How to get IoU in validation metrics?

### ISSUE THREAD

**Issue by vakker** [open-mmlab/mmdetection#7042]
Hi,

I'm running an instance segmentation model (`queryinst`), and the validation logs contain the following:
```
{"mode": "val", "epoch": 1, "iter": 100, "lr": 0.0, 
"bbox_mAP": 0.979, "bbox_mAP_50": 0.983, "bbox_mAP_75": 0.983, "bbox_mAP_s": -1.0, 
"bbox_mAP_m": -1.0, "bbox_mAP_l": 0.979, "bbox_mAP_copypaste": "0.979 0.983 0.983 -1.000 -1.000 0.979", 
"segm_mAP": 0.921, "segm_mAP_50": 0.983, "segm_mAP_75": 0.983, "segm_mAP_s": -1.0, 
"segm_mAP_m": -1.0, "segm_mAP_l": 0.921, "segm_mAP_copypaste": "0.921 0.983 0.983 -1.000 -1.000 0.921"}
```

How can I include additional metrics, like segm/bbox IoU in this?
Thanks

**Comment by RangiLyu** [comment]
COCO api does not support the `IOU` metric, it only supports bbox and instance segmentation mAP.

> My val set has 4772 elements, so why is the second val entry there in the middle of the validation iterations?
One of the 'val' is to evaluate, another 'val' is to post the losses on val dataset.

**Comment by vakker** [comment]
But the IOU calculation already exists in MMdet, as it's used for the mAP calculations (e.g. see [recall.py](https://github.com/open-mmlab/mmdetection/blob/ff9bc39913cb3ff5dde79d3933add7dc2561bab7/mmdet/core/evaluation/recall.py)), also there's IOU calculation in the training loss (e.g. `stage0_loss_iou`). So, I'm not sure how the COCO api is relevant here.

**Comment by RangiLyu** [comment]
Loss and evaluation metrics are two different concepts.

**Comment by vakker** [comment]
That's understood, but IOU is a metric commonly used for bbox and segm problems, and it would be trivial to add regardless of the COCO api, and in fact there's already an IOU calculation in the COCO evaluation, since it's required for the mAP calculations, right? So, would that be possible to report the bbox and segm (where applicable) IOU in the validation metrics?
